suite/tests/client-interface/drx_scattergather-test.c

```diff
@@ -0,0 +1,772 @@
+        movw       dx, 0xffff                              @N@\
+        kmovw      k1, edx                                 @N@\
+        vmovdqu32  xmm0, [REG_XAX]                         @N@\
+        opcode     [REG_XCX + xmm1 * 4] {k1}, xmm0         @N@\
+        kmovw      k1, edx                                 @N@\
+        vmovdqu32  ymm0, [REG_XAX + 16]                    @N@\
+        opcode     [REG_XCX + ymm1 * 4] {k1}, xmm0         @N@\
+        vmovdqu32  [REG_XCX + 16], ymm0                    @N@\
+        kmovw      k1, edx                                 @N@\
+        vmovdqu32  zmm0, [REG_XAX + 48]                    @N@\
+        opcode     [REG_XCX + zmm1 * 4] {k1}, ymm0         @N@\
+        add        REG_XSP, FRAME_PADDING                  @N@\
+        POP_CALLEE_SAVED_REGS()                            @N@\
+        ret                                                @N@\
+        END_FUNC(funcname)
+
+#undef OPCODE
+#define OPCODE vpscatterqd
+TEST_AVX512_SCATTER_IDX64_VAL32(FUNCNAME(OPCODE), OPCODE)
+#undef OPCODE
+#define OPCODE vscatterqps
+TEST_AVX512_SCATTER_IDX64_VAL32(FUNCNAME(OPCODE), OPCODE)
+
+#define TEST_AVX512_SCATTER_IDX64_VAL64(funcname, opcode)  @N@\
+DECLARE_FUNC_SEH(funcname)                                 @N@\
+  GLOBAL_LABEL(funcname:)                                  @N@\
+        mov        REG_XAX, ARG1                           @N@\
+        mov        REG_XCX, ARG3                           @N@\
+        mov        REG_XDX, ARG2                           @N@\
+        PUSH_CALLEE_SAVED_REGS()                           @N@\
+        sub        REG_XSP, FRAME_PADDING                  @N@\
+        END_PROLOG                                         @N@\
+        vmovdqu32  zmm1, [REG_XDX]                         @N@\
+        movw       dx, 0xffff                              @N@\
+        kmovw      k1, edx                                 @N@\
+        vmovdqu32  xmm0, [REG_XAX]                         @N@\
+        opcode     [REG_XCX + xmm1 * 4] {k1}, xmm0         @N@\
+        kmovw      k1, edx                                 @N@\
+        vmovdqu32  ymm0, [REG_XAX + 16]                    @N@\
+        opcode     [REG_XCX + ymm1 * 4] {k1}, ymm0         @N@\
+        vmovdqu32  [REG_XCX + 16], ymm0                    @N@\
+        kmovw      k1, edx                                 @N@\
+        vmovdqu32  zmm0, [REG_XAX + 48]                    @N@\
+        opcode     [REG_XCX + zmm1 * 4] {k1}, zmm0         @N@\
+        add        REG_XSP, FRAME_PADDING                  @N@\
+        POP_CALLEE_SAVED_REGS()                            @N@\
+        ret                                                @N@\
+        END_FUNC(funcname)
+
+#undef OPCODE
+#define OPCODE vpscatterqq
+TEST_AVX512_SCATTER_IDX64_VAL64(FUNCNAME(OPCODE), OPCODE)
+#undef OPCODE
+#define OPCODE vscatterqpd
+TEST_AVX512_SCATTER_IDX64_VAL64(FUNCNAME(OPCODE), OPCODE)
+#endif /* __AVX512F__ */
```

Comment:
suggest: banner comment to help identify these different regions: avx512 vs avx2

Reply:
Done.